context/src/main/java/io/opentelemetry/context/Scope.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package io.opentelemetry.context
+
+/**
+ * An [AutoCloseable] that represents a mounted context for a block of code. A failure to call
+ * [Scope.close] will generally break tracing or cause memory leaks. It is recommended that
+ * you use this class with a `try-with-resources` block:
+ *
+ * <pre>`try (Scope ignored = span.makeCurrent()) {
+ * ...
+ * }
+`</pre> *
+ */
+interface Scope : AutoCloseable {
```

Comment:
add open question: figure out syntax for kotlin, autocloseable vs closeable, `using`?